### PR TITLE
Theme Config Helper Edge fixes

### PR DESCRIPTION
**Pull request by crhallberg**
- Fixes a bug when string values were being merged (still returns array)
- Adds ability to return entire config.

- [x] Add caching? Move caching from ThemeConfig helper up to ThemeInfo?

### CODE REVIEW

File: module/VuFindTheme/tests/unit-tests/src/VuFindTest/ThemeInfoTest.php [hunk 1]
```diff
@@ -310,6 +310,30 @@ public function testGetMergedConfigMixinWithFlattening()
         );
     }
 
+    /**
+     * Test getMergedConfig() with string return
```
Comment (demiankatz) [inline module/VuFindTheme/tests/unit-tests/src/VuFindTest/ThemeInfoTest.php]: Just curious about name/comment on this test -- not sure what you mean by "with string return" since this is returning an array. Do you mean with string key?

**Comment by demiankatz** [comment]
Thanks, @crhallberg -- apart from the minor question above, this makes sense to me. I do agree that it might make more sense to move the caching into ThemeInfo instead of the helper, since then you'll be caching data at the source.... If you were to do that, though, it would be a good idea to make the cache an optional dependency, and only cache if it's injected, so we don't introduce a backward-breaking change in a minor release.

**Comment by demiankatz** [comment]
Thanks for the comment cleanup, @crhallberg. Do you want me to merge this now, or are you still contemplating the cache changes?

File: module/VuFind/src/VuFind/View/Helper/Root/ThemeConfigFactory.php [hunk 2]
```diff
@@ -69,15 +69,16 @@ public function __invoke(
...
         // As of release 1.1.0, the memory storage adapter has a flaw which can cause
         // unnecessary out of memory exceptions when a memory limit is enabled; we
         // can disable these problematic checks by setting memory_limit to -1.
         $cacheConfig = ['name' => 'memory', 'options' => ['memory_limit' => -1]];
         $cache = $container->get(\Laminas\Cache\Service\StorageAdapterFactory::class)
             ->createFromArrayConfiguration($cacheConfig);
 
-        return new $requestedName($themeInfo, $cache);
+        $themeInfo = $container->get(\VuFindTheme\ThemeInfo::class);
+        $themeInfo->setCache($cache);
```
Comment (demiankatz) [inline module/VuFind/src/VuFind/View/Helper/Root/ThemeConfigFactory.php]: I think it would make much more sense to set the cache in the factory for ThemeInfo rather than here. We don't want the factory for one service to modify a different service as a side effect.

File: module/VuFindTheme/src/VuFindTheme/ThemeInfo.php [hunk 3]
```diff
@@ -176,7 +197,7 @@ protected function loadThemeConfig($theme)
     public function getThemeInfo()
     {
         // Fill in the theme info cache if it is not already populated:
-        if (null === $this->allThemeInfo) {
+        if ($this->allThemeInfo === null) {
```
Comment (demiankatz) [inline module/VuFindTheme/src/VuFindTheme/ThemeInfo.php:200]: Any particular reason you reversed the comparison here? Seems like a potentially unnecessary diff.

File: module/VuFindTheme/src/VuFindTheme/ThemeInfo.php [hunk 4]
```diff
@@ -82,6 +91,18 @@ public function __construct($baseDir, $safeTheme)
...
     }
 
+    /**
+     * Provide cache and activate info caching
+     *
+     * @param StorageInterface $cache cache object
+     *
+     * @return void
+     */
+    public function setCache(StorageInterface $cache)
```
Comment (demiankatz) [inline module/VuFindTheme/src/VuFindTheme/ThemeInfo.php:101]: In order to keep our coverage up, I think we should add some new tests to [ThemeInfoTest](https://github.com/vufind-org/vufind/blob/dev/module/VuFindTheme/tests/unit-tests/src/VuFindTest/ThemeInfoTest.php) to inject mock cache objects to confirm that values coming from the cache get returned and that the cache gets updated when it returns null.

**Review by demiankatz — CHANGES_REQUESTED** [review]
Thanks for the progress, @crhallberg -- I think we need just a little more work done here.

Also, do you have any idea why the Snyk service is failing the build? I think that's something you set up. Has it changed or broken? Do we need to remove it, or can it be fixed?

File: module/VuFindTheme/src/VuFindTheme/ThemeInfo.php (hunk 3 above)
Comment (crhallberg) [reply module/VuFindTheme/src/VuFindTheme/ThemeInfo.php:200]: Just a nitpick.

Comment (demiankatz) [reply module/VuFindTheme/src/VuFindTheme/ThemeInfo.php:200]: I thought the reversed form was preferred since it prevents accidental assignments, but it's definitely not a hill I'm willing to die on. ;-)

**Review by demiankatz — APPROVED** [review]
I made a couple of trivial tweaks (added a test assertion and updated a comment). I think this is ready to go now. Thanks!
